mysql/datadog_checks/mysql/statements.py

```diff
@@ -60,38 +63,33 @@
 }
 
 
-def generate_synthetic_rows(rows):
-    # type: (List[PyMysqlRow]) -> List[PyMysqlRow]
-    """
-    Given a list of rows, generate a new list of rows with "synthetic" column values derived from
-    the existing row values.
-    """
-    synthetic_rows = []
-    for row in rows:
-        new = copy.copy(row)
-        new['avg_time'] = float(new['time']) / new['count'] if new['count'] > 0 else 0
-        new['rows_sent_ratio'] = float(new['rows_sent']) / new['rows_examined'] if new['rows_examined'] > 0 else 0
-
-        synthetic_rows.append(new)
-
-    return synthetic_rows
-
-
 class MySQLStatementMetrics(object):
     """
     MySQLStatementMetrics collects database metrics per normalized MySQL statement
     """
 
-    def __init__(self, config):
+    def __init__(self, check, config):
         # type: (MySQLConfig) -> None
-        self.config = config
+        self._check = check
+        self._config = config
+        self._db_hostname = resolve_db_host(self._config.host)
         self.log = get_check_logger()
         self._state = StatementMetrics()
 
-    def collect_per_statement_metrics(self, db):
-        # type: (pymysql.connections.Connection) -> List[Metric]
+    def collect_per_statement_metrics(self, db, tags):
+        # type: (pymysql.connections.Connection, List[string]) -> None
         try:
-            return self._collect_per_statement_metrics(db)
+            rows = self._collect_per_statement_metrics(db)
+            if not rows:
+                return
+            payload = {
+                'host': self._db_hostname,
+                'timestamp': time.time() * 1000,
+                'min_collection_interval': self._config.min_collection_interval,
+                'tags': tags,
+                'mysql_rows': rows,
+            }
+            self._check.database_monitoring_query_metrics(json.dumps(payload, default=default_json_event_encoding))
         except Exception:
             self.log.exception('Unable to collect statement metrics due to an error')
             return []
```

Comment:
for consistent return type, maybe remove this?

Reply:
Good catch, removed